### PR TITLE
add circle ci

### PR DESCRIPTION
**Pull request by gengjiawen**
But looks like there are some failing test.

see https://circleci.com/gh/gengjiawen/RxPermission

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+          command: emulator -avd testEmulator -no-audio -no-window
+          background: true
+      - run:
+          name: Wait for emulator
+          command: chmod +x ./android-wait-for-boot.sh && ./android-wait-for-boot.sh
+      - run:
+          name: Remove lock screen
+          command: adb shell input keyevent 82
+      - run:
+          name: Build
```
Comment (vanniktech) [inline circle.yml]: We could let Travis do that and use Circle CI only for Integration Tests.

File: circle.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+          command: ./gradlew build
+      - run:
+          name: Unit testing
```
Comment (vanniktech) [inline circle.yml]: We could let Travis do that and use Circle CI only for Integration Tests.

**Comment by vanniktech** [comment]
Try updating your branch with the latest master.

**Comment by gengjiawen** [comment]
api-19 ?

**Comment by vanniktech** [comment]
@gengjiawen I've just run them again on a different machine with `@Nexus_4_API_28` and they do pass locally for me. I think Circle CI is just flaky.

**Comment by vanniktech** [comment]
Feel free to abandon this. I've also tried many many times getting the emulator to work on CI but it never did - which is sad.

**Comment by vanniktech** [comment]
I do appreciate your effort @gengjiawen

**Pull request closed without merging**